transformations/suspecting_paraphraser/transformation.py

```diff
@@ -0,0 +1,192 @@
+from typing import Tuple, List
+import nltk
+from pyinflect import getInflection
+import spacy
+import gender_guesser.detector as gender
+from enum import Enum
+import numpy as np
+
+
+from interfaces.QuestionAnswerOperation import QuestionAnswerOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+class Pronouns:
+    HE = 0
+    SHE = 1
+    IT = 2
+    THEY = 3
+
+
+class SuspectingParaphraser(QuestionAnswerOperation):
+    tasks = [TaskType.QUESTION_ANSWERING, TaskType.QUESTION_GENERATION]
+
+    languages = ["en"]
+
+    def __init__(self, seed=0, max_output=1, pronoun_mod=0.9):
+        super().__init__(seed)
+        np.random.seed(seed)
+        nltk.download("punkt")
+
+        self.seed = seed
+        self.max_output = max_output
```

Comment:
Duplicate code: Better to use 'max_outputs' present in the parent class and initialize it via super call.